src/client/dfs/dfs.c

```diff
@@ -1613,6 +1607,99 @@ dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 	return rc;
 }
 
+int
+dfs_lookup_rel(dfs_t *dfs, dfs_obj_t *parent, const char *name, int flags,
+	       dfs_obj_t **_obj, mode_t *mode)
+{
+	dfs_obj_t		*obj;
+	struct dfs_entry	entry = {0};
+	bool			exists;
+	int			daos_mode;
+	int			rc = 0;
+
+	if (dfs == NULL || !dfs->mounted)
+		return -DER_INVAL;
+	if (_obj == NULL)
+		return -DER_INVAL;
+	if (parent == NULL)
+		parent = &dfs->root;
```

Comment:
Is there any case you'd want this rather than returning -DER_INVAL?

Reply:
you mean using the root? yes, for MPI-IO now we assume DFS mount is just flat, and all parents passed are NULL.
I think this would change in the future to allow nested dirs, in which case we can change that to return inval.

Reply:
Does MPI-IO use the new dfs_lookup_rel() API call though, or are you planning on removing dfs_lookup() and migrating it?

Reply:
considering that this is not merged yet, no it doesn't use it :-)
I'm just being consistent with behavior of the other APIs.

Reply:
Ah yes, I see the rest are like it.  It seems like a dangerous default, as it could result in a software but doing the wrong thing rather than failing correctly however as you say this is consistent with other functions.

Reply:
right, i think we do need to revisit this. but i just wanted to avoid a lookup call in a flat namespace. the solution would have to be like a special DFS mount called flat, that allows this but fails for others. something to add to the bucket list of TBD.